### PR TITLE
[INLONG-9143][Agent] Add log file collect task

### LINKED ISSUE #9143: [Improve][Agent] Add log file collect task
### Description

Add log file collect task

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
### Prepare a Pull Request
[INLONG-9143][Agent] Add log file collect task

- Fixes #9143 

### Motivation

Add log file collect task

### Modifications

Add log file collect task

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/AgentErrMsg.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.inlong.agent.plugin.task.filecollect;
+
+public class AgentErrMsg {
+
+    public static final String CONFIG_SUCCESS = "SUCCESS";
+
+    // 数据源配置异常 */
```
Comment (doleyzi) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/AgentErrMsg.java:24]: chinese

**Review by doleyzi — APPROVED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/AgentErrMsg.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    public static final String FILE_OP_ERROR = "ERROR-1-TDAgent|30002|ERROR|ERROR_OPERATE_FILE|";
+
+    // 磁盘满
+    public static final String DISK_FULL = "ERROR-1-TDAgent|30001|FATAL|FATAL_DISK_FULL|";
+
+    // 内存溢出
+    public static final String OOM_ERROR = "ERROR-1-TDAgent|30001|FATAL|FATAL_OOM_ERROR|";
+
+    // watcher异常
+    public static final String WATCHER_INVALID = "ERROR-1-TDAgent|40001|WARN|WARN_INVALID_WATCHER|";
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/AgentErrMsg.java:49]: Tdagent

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/AgentBaseTestsHelper.java [hunk 3]
```diff
@@ -71,4 +77,32 @@ public void teardownAgentHome() {
...
+        TaskProfile profile = TaskProfile.convertToTaskProfile(dataConfig);
+        return profile;
+    }
+
+    private DataConfig getDataConfig(int taskId, String pattern, boolean retry, Long startTime, Long endTime,
+            TaskStateEnum state) {
+        DataConfig dataConfig = new DataConfig();
+        dataConfig.setInlongGroupId("testGroupId"); // 老字段 groupId
+        dataConfig.setInlongStreamId("testStreamId"); // 老字段 streamId
+        dataConfig.setDataReportType(1); // 老字段 reportType
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/AgentBaseTestsHelper.java:93]: chinese

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/WatchEntity.java [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+                if (!pathToKeys.containsKey(dirName) && match(pattern, dirName)) {
+                    try {
+                        WatchKey key = dirPath
+                                .register(watchService, StandardWatchEventKinds.ENTRY_CREATE);
+                        keys.put(key, dirPath);
+                        pathToKeys.put(dirName, key);
+                        logger.info("Register a new directory: " + dirName);
+                    } catch (IOException e) {
+                        /**
+                         * 捕获异常，不能注册的子目录就忽略。
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/WatchEntity.java:201]: chinese

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/AgentErrMsg.java (hunk 1 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/AgentErrMsg.java:24]: get

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/FileScanner.java [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+        ArrayList<String> allPaths = FilePathUtil.cutDirectory(filename);
+        String firstDir = allPaths.get(0);
+        String secondDir = allPaths.get(0) + File.separator + allPaths.get(1);
+        return getUpdatedOrNewFiles(firstDir, secondDir, filename, 3, maxFileNum);
+    }
+
+    private static ArrayList<String> getUpdatedOrNewFiles(String firstDir, String secondDir,
+            String fileName, long depth, int maxFileNum) {
+
+        // logger.info("getUpdatedOrNewFiles: firstdir: {}, seconddir: {} filename: {}",
```
Comment (doleyzi) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/FileScanner.java:118]: It is recommended to delete the commented code

**Review by doleyzi — APPROVED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/AgentErrMsg.java (hunk 2 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/AgentErrMsg.java:49]: get

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/FileScanner.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/*
+ * This class is mainly used for scanning log file that we want to read. We use this class at
+ * tdagent recover process, the do and redo tasks and the current log file access when we deploy a
```
Comment (doleyzi) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/FileScanner.java:40]: tdagent

**Review by doleyzi — APPROVED** [review]

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/AgentBaseTestsHelper.java (hunk 3 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/AgentBaseTestsHelper.java:93]: get

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/WatchEntity.java (hunk 4 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/WatchEntity.java:201]: get

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/FileScanner.java (hunk 5 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/FileScanner.java:118]: get

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/FileScanner.java (hunk 6 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/FileScanner.java:40]: get

**Review by doleyzi — APPROVED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/AgentErrMsg.java (hunk 1 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/AgentErrMsg.java:24]: fix in next issue

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/AgentErrMsg.java (hunk 2 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/AgentErrMsg.java:49]: fix in next issue

**Review by EMsnap — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

**Pull request merged**